backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +75,127 @@ def disambiguate_signatures(self, clusters):
     db.session.commit()
 
 
+def match_literature_author_with_config(author_data, matcher_config):
+    matched_records = [
+        matched_record
+        for matched_record in match(author_data, matcher_config)
+        if matched_record.get("inner_hits")
+    ]
+    matched_authors_refs = match_literature_author_ids(matched_records)
+    return matched_authors_refs, matched_records
```

Comment:
return type is a bit awkward, I don't think the first line needs to be here. Can't you put it in something like `assign_reference_to_author_if_unambiguous_literature_author_match`, which would call this then the existing `assign_reference_to_author_if_unambiguous_match`?